README.md

```diff
@@ -6,8 +6,63 @@ If you have any feedback or are interested in what we are doing please [contact
 
 API for managing and accessing Ably Spaces.
 
+**Class definitions are available [here](/docs/class-definitions.md).**
+
 ## Spaces
 
 A Space is an Ably concept that allows for shared realtime digital experiences such as chat, reactions and collaborative editing.
 
 A Space consists of users and storage with persistent presence.
+
+### Usage
+
+To use the Space API, you'll need to have an Ably client already set up and authenticated.
+To get started quickly, you can use [basic authentication](https://ably.com/docs/realtime/authentication#basic-authentication)
+by creating an API key with the **publish**, **subscribe** and **presence** capabilities enabled.
+
+
+Here's an example of how you can create a new instance of the client and pass it to the Spaces constructor:
+
+```ts
+import { Realtime } from 'ably/promise';
+import Spaces from '@ably-labs/spaces';
+
+const client = new Realtime(ABLY_API_KEY);
+const spaces = new Spaces(client);
+```
+
+A new space can then be created and entered:
+
+```ts
+const space = spaces.get("mySpace");
+space.enter({
+  username: "Example User",
+  avatar: "https://example.com/users/1234.png",
+});
+```
+
+Enter and leave events can be listened to by creating an event listener:
+
+```ts
+space.on('membersUpdate', (members) => {
```

Comment:
Is it worth having explicit, type-safe methods rather than relying on passing strings to `space.on` (e.g. `space.onMembersUpdate`)?

Reply:
Does the plural event name (i.e. `membersUpdate`) imply that all space members are passed to the callback when any one of them have updated? If so, then that seems like an odd API, since how will you know which of the members actually updated? And if this only gets called with the member that was updated, should it not be singular rather than plural?

Reply:
@lmars this API was designed based on what kind of UI it's supposed to support. In this case, it's a collection of members that are likely to be displayed as some sort of list. We manage that list, so the UI JavaScript can just re-render that list. This might seem odd to you, but it's pretty standard fare in modern UI development, where you pass data to your renders which decide what's the most efficient way to update the screen.

We might have a singular equivalent at a later date, if we find the right use case.

Reply:
> it's a collection of members that are likely to be displayed as some sort of list

Ok thanks, so will the callback always be called with the complete list of members in the space each time any of them enter/leave/update? If so, I do think the description should make that a little more explicit (e.g. "The list of space members changes when members enter or leave the space. Use an event listener to listen to changes to the list:")

Reply:
> Ok thanks, so will the callback always be called with the complete list of members in the space each time any of them enter/leave/update

that's correct.

>I do think the description should make that a little more explicit

agree, that does make it more clear